components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/widget/components/CustomTimeRangeSelector.jsx

```diff
@@ -0,0 +1,324 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import Moment from "moment";
+import TimePicker from "./TimePicker";
+import Button from "@material-ui/core/Button/Button";
+import Typography from "@material-ui/core/Typography/Typography";
+export default class CustomTimeRangeSelector extends React.Component {
+  state = {
+    invalidDateRange: false,
+    customRangeGranularityValue: this.props.customRangeGranularityValue,
+    startTime: this.props.startTime,
+    endTime: new Date(),
+    applyButtonBackgroundColor: "#ef6c00",
+    cancelButtonBackgroundColor: "#999"
+  };
+
+  /**
+   * Setting the start time using the custom  ranges
+   * @param{object} date
+   */
+  handleStartTimeChange = (date) => {
+    const { endTime } = this.state;
+    const startTime = date;
+    if (Moment(startTime).isSameOrAfter(Moment(endTime))) {
+      this.setState({
+        invalidDateRange: true,
+        applyButtonBackgroundColor: "#999"
+      });
+    } else {
+      this.setState({
+        invalidDateRange: false,
+        applyButtonBackgroundColor: "#ef6c00",
+        startTime: date
+      });
+    }
+  };
+  /**
+   * Setting the end time using the custom  ranges
+   * @param{object} date
+   */
+  handleEndTimeChange = (date) => {
+    const { startTime } = this.state;
+    const endTime = date;
+    if (Moment(startTime).isSameOrAfter(Moment(endTime))) {
+      this.setState({
+        invalidDateRange: true,
+        applyButtonBackgroundColor: "#999"
+      });
+    } else {
+      this.setState({
+        invalidDateRange: false,
+        applyButtonBackgroundColor: "#ef6c00",
+        endTime: date
+      });
+    }
+  };
+
+  /*Publishing the custom time range
+  onChangeCustom()=>handleGranularityChangeForCustom(mode, startTime, endTime, granularity)
+  mode:custom
+  granularity:second,minute,hour,day,month,year
+  */
+  publishCustomTimeRange = () => {
+    const { handleClose, onChangeCustom } = this.props;
+    const { customRangeGranularityValue, startTime, endTime } = this.state;
+    handleClose();
+    onChangeCustom("custom", startTime, endTime, customRangeGranularityValue);
+  };
+
+  /**
+   * Change the granularity value of the custom ranges. View of the custom ranges will
+   * change according to the granularity value
+   * @param{string} customGranularityValue:'second','minute','hour' etc
+   */
+  changeCustomRangeGranularity = (customGranularityValue) => {
+    this.props.disableSelectedQuickRangeValue()
+    const { startTime, endTime } = this.state;
+    let invalidDateRange = false;
+    let applyBtnColor = "#ef6c00";
+    switch (customGranularityValue) {
```

Comment:
No need of this switch statement. Instead can be done as follows.

```js
if (Moment(startTime).isSameOrAfter(Moment(endTime), customGranularityValue)) {
    invalidDateRange = true;
    applyBtnColor = "#999";
}
```

Reply:
In here according to the granularity value we select in the custom time range, the time format need to be change.
As as example
when the initial time( when the pop up shows) = 3 
months it will stored as 2019 Jan 01 10:00:00 AM to 2019 April 01 10:00:00
when user selects a granularity it the final timestamp will vary,like
when user selects year granularity in custom ranges it needed to show only years and cannot allow user to apply the range from 2019 to 2019. that's why I used a switch statement to filter the necessary time format and allow or disallow the apply button